<launch>
  <arg name="output_screen" default="false"/>
  <arg name="image_topic" default="image" />
  <arg name="camera_info_topic" default="camera_info" />
  <arg name="debug_view" default="false"/>
  <arg name="object_height" default="0.2"/>
  <arg name="contour_area_size" default="0.05"/>
  <arg name="contour_area_margin" default="0.01"/>
  <arg name="frame_id" default="target_object"/>
  <arg name="h_limit_max" default="33" doc="The maximum allowed field value Hue" />
  <arg name="h_limit_min" default="10" doc="The minimum allowed field value Hue" />
  <arg name="s_limit_max" default="255" doc="The maximum allowed field value Saturation" />
  <arg name="s_limit_min" default="0" doc="The minimum allowed field value Saturation" />
  <arg name="v_limit_max" default="255" doc="The maximum allowed field value Value" />
  <arg name="v_limit_min" default="0" doc="The minimum allowed field value Value" />

  <node pkg="nodelet" type="nodelet" name="object_detection_nodelet" args="manager" output="screen" if="$(arg output_screen)"/>
  <node pkg="nodelet" type="nodelet" name="object_detection_nodelet" args="manager" unless="$(arg output_screen)"/>

  <node pkg="nodelet" type="nodelet" name="hsv_color_filter"
        args="load opencv_apps/hsv_color_filter object_detection_nodelet">
    <remap from="image" to="$(arg image_topic)"/>
    <param name="debug_view" value="false"/>
    <param name="h_limit_max" value="$(arg h_limit_max)"/>
    <param name="h_limit_min" value="$(arg h_limit_min)"/>
    <param name="s_limit_max" value="$(arg s_limit_max)"/>
    <param name="s_limit_min" value="$(arg s_limit_min)"/>
    <param name="v_limit_max" value="$(arg v_limit_max)"/>
    <param name="v_limit_min" value="$(arg v_limit_min)"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="ground_object_detection_with_size_filter"
        args="load aerial_robot_perception/GroundObjectDetectionWithSizeFilter object_detection_nodelet" output="screen">
    <remap from="image" to="/hsv_color_filter/image" />
    <remap from="camera_info" to="$(arg camera_info_topic)"/>
    <param name="object_height" value="$(arg object_height)"/>
    <param name="contour_area_size" value="$(arg contour_area_size)"/>
    <param name="contour_area_margin" value="$(arg contour_area_margin)"/>
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
  </node>

</launch>
